pkg/osd_cluster_upgrader/upgrader.go

```diff
@@ -247,6 +248,27 @@ func AllWorkersUpgraded(c client.Client, scaler scaler.Scaler, metricsClient met
 		return false, err
 	}
 
+	silenceActive, err := metricsClient.IsSilenceActive()
+	if err != nil {
+		return false, err
+	}
+
+	var nodeCompleteTime *metav1.Time
+	for _, uc := range upgradeConfig.Status.History {
+		if uc.Version == upgradeConfig.Spec.Desired.Version {
+			for _, ucc := range uc.Conditions {
+				if ucc.Type == "AllWorkerNodesUpgraded" {
+					nodeCompleteTime = ucc.CompleteTime
+				}
+			}
+		}
+	}
+
+	if !silenceActive && nodeCompleteTime == nil {
```

Comment:
Can we do node upgrade timeout per node ?

Reply:
We already had the logic to setup the silence based on the node number.
As discussed with john, it is more easier to use the existing silence.

Reply:
per node? Do you mean check the timeout for every node separately?

Reply:
@bmeng yes, I know it's not easy, but we should have that, like if we have 10 nodes, and upgrade node stuck at first one, we should got alert if first node upgrade timeout instead of 10 nodes timeout, WDYT?

Reply:
Yeah, it is not quite easy IMO. Seems like the nodes will be upgraded parallel? 

Another thing is, we setup the silence for the warning and lower severity. 
As the condition you mentioned, if the node stuck, I think some of the operators will be Degraded. 
So we will get paged regardless of the upgrade, but the cluster healthy.

Reply:
I think we will eventually need to refine this watch per node but for M1 this is ok

Reply:
I think we will eventually need to refine this watch per node but for M1 this is ok

Reply:
Sure, @bmeng Could you create one ticket to track this ?